### PR TITLE
feat: Add GBFS feeds to the database

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

This PR introduces the capability to add GBFS feeds to the database.

**Expected Behavior:**

The GBFS feeds are now integrated into the database as `Feed` and `GbfsFeed` entities. Each feed includes location information, external IDs referencing the `System ID` from `systems.csv`, and associated `GbfsVersion` entries. To avoid serving GBFS feeds temporarily, the `gbfs` data types have been filtered out of the `feeds` endpoint.

image

**Out of Scope:** 

This PR does not include the addition of a trigger for updates from the `gbfs` repository. For more details, refer to https://github.com/MobilityData/mobility-feed-api/issues/675.

**Testing Tips:**
- Checkout the `feat/565-2` branch.
- Initialize the local database.
- Download `systems.csv` from the `gbfs` repository.
- Install dependencies and run: `python3 populate_db_gbfs.py --filepath "<path to systems.csv>"`.
- Validate the database content to ensure the feeds are correctly added.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by cka-y** [comment]
The integration tests are expected to fail with status code 500 since the database schema in QA does not match the database changes introduced by this PR

File: liquibase/changes/feat_565.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+ALTER TYPE datatype ADD VALUE 'gbfs';
+
+CREATE TABLE GBFS_Feed(
```
Comment (davidgamez) [inline liquibase/changes/feat_565.sql:3]: [picky]: Just for consistency name the table `GBFSFeed` like `GTFSFeed`. The Model name is generated as `GbfsFeed` vs `Gbfsfeed`. I agree that this proposal is a better naming convention; However is different that the current one in addition the Model generated `GBFSFeed` has the same pattern than the API model that would have been generated(not currently as it's not defined in the OpenApiSchema)

File: liquibase/changes/feat_565.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    id VARCHAR(255) PRIMARY KEY,
+    operator VARCHAR(255),
+    operator_url VARCHAR(255),
+    auto_discovery_url VARCHAR(255),
+    FOREIGN KEY (id) REFERENCES Feed(id)
+);
+
+CREATE TABLE GBFS_Version(
```
Comment (davidgamez) [inline liquibase/changes/feat_565.sql:11]: The same comment Re: naming convention.

**Review by davidgamez — APPROVED** [review]

File: liquibase/changes/feat_565.sql (hunk 1 above)
Comment (cka-y) [reply liquibase/changes/feat_565.sql]: Agreed! I updated the naming

**Comment by cka-y** [comment]
Opened #678 to address potential issues related to adding GBFS feeds to the database without corresponding updates to the API.

File: api/src/scripts/populate_db_gbfs.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import pytz
+
+from database.database import generate_unique_id, configure_polymorphic_mappers
+from database_gen.sqlacodegen_models import Gbfsfeed, Location, Gbfsversion, Externalid
+from scripts.gbfs_utils.comparison import generate_system_csv_from_db, compare_db_to_csv
+from scripts.gbfs_utils.fetching import fetch_data, get_data_content, get_gbfs_versions
+from scripts.gbfs_utils.license import get_license_url
+from scripts.populate_db import DatabasePopulateHelper, set_up_configs
+
+OFFICIAL_VERSIONS = [
```
Comment (davidgamez) [inline api/src/scripts/populate_db_gbfs.py:13]: I'm not sure about the best approach for this, but this list needs to be more visible and easy to spot when a new GBFS version is released. Without over-architecting a complex solution, adding a documentation section to the README file in this repository may help users find the list when needed.

Comment (cka-y) [reply api/src/scripts/populate_db_gbfs.py:13]: Not sure the readme of our repo is the appropriate place for this information taking into account that we are not serving gbfs feeds through the api yet

**Review by davidgamez — APPROVED** [review]

**Comment by richfab** [comment]
Thank you for the ping @cka-y. Great work!

**Pull request merged**